compiler/rustc_lint/src/hidden_unicode_codepoints.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{EarlyContext, EarlyLintPass, LintContext};
+use rustc_ast as ast;
+use rustc_errors::{Applicability, SuggestionStyle};
+use rustc_span::{BytePos, Span, Symbol};
+
+declare_lint! {
+    /// The `text_direction_codepoint_in_literal` lint detects Unicode codepoints that change the
+    /// visual representation of text on screen in a way that does not correspond to their on
+    /// memory representation.
+    ///
+    /// ### Explanation
+    ///
+    /// The unicode characters `\u{202A}`, `\u{202B}`, `\u{202D}`, `\u{202E}`, `\u{2066}`,
+    /// `\u{2067}`, `\u{2068}`, `\u{202C}` and `\u{2069}` make the flow of text on screen change
+    /// its direction on software that supports these codepoints. This makes the text "abc" display
+    /// as "cba" on screen. By leveraging software that supports these, people can write specially
+    /// crafted literals that make the surrounding code seem like it's performing one action, when
+    /// in reality it is performing another. Because of this, we proactively lint against their
+    /// presence to avoid surprises.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// #![deny(text_direction_codepoint_in_literal)]
+    /// fn main() {
+    ///     println!("{:?}", '‮');
```

Comment:
How is this able to compile without an `#[allow]`? I've thought the lint also forbids usage in doc comments? Or are examples exempt?

Reply:
Interesting... The test itself is a `compile_fail`, but you're saying it should fail at the library level?

Reply:
@est31 is saying that the lint should trigger when compiling `hidden_unicode_codepoints.rs` because it appears in a doc-comment. Now that it is pointed out, clearly there's something amiss as well here.

Reply:
It's been pointed out on the main PR as well (should have checked that PR before making the comment): https://github.com/rust-lang/rust/pull/90460#pullrequestreview-793969130